src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -884,6 +884,14 @@ public static decimal Max(decimal val1, decimal val2)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static double Max(double val1, double val2)
         {
+            if (AdvSimd.Arm64.IsSupported &&
+                (!RuntimeHelpers.IsKnownConstant(val1) || !RuntimeHelpers.IsKnownConstant(val1)))
```

Comment:
Should these conditions be:
```suggestion
                (!RuntimeHelpers.IsKnownConstant(val1) || !RuntimeHelpers.IsKnownConstant(val2)))
```